tests/api/test_api_v2.py

```diff
@@ -128,5 +130,168 @@ def test_get_component_analyses_with_result_not_none(
         result = 'my_package_analyses_result'
         _vendor_analyses.return_value = result
         ca = ComponentAnalysesApi()
-        analyses_result = ca.get("npm", "pkg", "ver")
+        analyses_result = ca.get('npm', 'pkg', 'ver')
         self.assertEqual(analyses_result, result)
+
+
+class TestStackAnalysesGetApi(unittest.TestCase):
+    """Stack Analyses Unit Tests."""
+
+    @patch('bayesian.api.api_v2.RdbAnalyses.get_request_data', return_value=None)
+    def test_sa_get_with_invalid_id(self, _request_data):
+        """Get endpoint with invalid request id to get NONE data from db."""
+        sa = StackAnalysesApi()
+        self.assertRaises(HTTPError, sa.get, 'request_id')
+
+    @patch('bayesian.api.api_v2.RdbAnalyses.get_request_data', return_value={})
+    @patch('bayesian.api.api_v2.RdbAnalyses.get_stack_result', return_value={})
+    @patch('bayesian.api.api_v2.RdbAnalyses.get_recommendation_data', return_value={})
+    @patch('bayesian.api.api_v2.StackAnalysesResponseBuilder.get_response')
+    def test_sa_get_request_success(self, _get_response, _recommendation_data, _stack_result,
```

Comment:
I see that excessive mocking in the tests. It makes harder to refactor the code. I suggest to do mocking only for the external component(in this case RDS). So the recommendation is to "Mock across architecturally significant boundaries, but not within those boundaries".

Reply:
Fixed in commit 'Added dependecy injection for RDB class from API'

Reply:
I still see excessive mocking. In this test case, it seems to be valid to mock only get_request_data & get_stack_result, nothing else should be mocked.